model/project_ref.go

```diff
@@ -2203,6 +2250,38 @@ func (p *ProjectRef) GetGithubProjectConflicts() (GithubProjectConflicts, error)
 	return res, nil
 }
 
+// ValidateProjectCreation returns an error if the total or owner/repo project limit set by the admin settings
+// has been reached. If the project is not enabled, it will grip a warning but not error.
+func ValidateProjectCreation(config *evergreen.Settings, projectRef *ProjectRef) error {
+	catcher := grip.NewBasicCatcher()
+	allEnabledProjects, err := GetNumberOfEnabledProjects()
+	if err != nil {
+		return errors.Wrap(err, "getting number of projects")
+	}
+	if allEnabledProjects >= config.ProjectCreation.TotalProjectLimit {
+		catcher.Errorf("total project limit of %d reached", config.ProjectCreation.TotalProjectLimit)
+	}
+	enabledOwnerRepoProjects, err := GetNumberOfEnabledProjectsForOwnerRepo(projectRef.Owner, projectRef.Repo)
+	if err != nil {
+		return errors.Wrapf(err, "getting number of projects for '%s'/'%s'", projectRef.Owner, projectRef.Repo)
+	}
+	if !config.ProjectCreation.IsExceptionToRepoLimit(projectRef.Owner, projectRef.Repo) && enabledOwnerRepoProjects >= config.ProjectCreation.RepoProjectLimit {
```

Comment:
In another thread I mentioned checking exception before counting projects to avoid a query; making a new thread since that one was outdated but still needs to be addressed

Reply:
done